lib/agent.js

```diff
@@ -47,17 +47,26 @@ class Agent extends EventEmitter {
       throw new InvalidArgumentError('Origin must be a non-empty string.')
     }
 
-    let pool = this[kCache].get(origin)
+    return this[kFactory](origin)
```

Comment:
this should return the same instance that is being cached.

Reply:
Good catch!